src/audio/eq_iir/eq_iir.c

```diff
@@ -652,21 +657,31 @@ static int iir_cmd_get_data(struct comp_dev *dev,
 
 		/* Copy back to user space */
 		if (cd->config) {
+			src = (unsigned char *)cd->config;
+			dst = (unsigned char *)cdata->data->data;
 			bs = cd->config->size;
-			comp_info(dev, "iir_cmd_set_data(), size %u",
-				  bs);
-			if (bs > SOF_EQ_IIR_MAX_SIZE || bs == 0 ||
-			    bs > max_size)
-				return -EINVAL;
-			ret = memcpy_s(cdata->data->data,
-				       ((struct sof_abi_hdr *)
-				       (cdata->data))->size, cd->config, bs);
+			cdata->elems_remaining = 0;
+			offset = 0;
+			if (bs > max_size) {
```

Comment:
what are we doing here ? maybe some comments needed.

Reply:
Good that you asked! I copied this from FIR and it seems there's more change need (both components). I seems that the get_data() for multi-part happens only for a topology embedded long blob and I didn't test it, I tested before this only with sof-ctl. There seems to be a similar memcpy_s() assert related panic happening that I need to fix. The set_data() worked OK as far as I see prior to crash.

Reply:
I think I found the mistake with size. I'll update this PR tomorrow.

Reply:
I had from above the headers size subtraction missing from max_size. It caused the assert() in previous PR version. Now both topology embedding and sof-ctl ways for setup work. I've added comments to code to understand what happens.